pkgs/servers/nextcloud/packages/default.nix

```diff
@@ -26,4 +32,5 @@ let packages = self:
     lib.makeExtensible (_: lib.mapAttrs (pname: data: self.mkNextcloudDerivation { inherit pname; inherit data; }) pkgs))
     generatedJson;
 
-in (lib.makeExtensible (_: (lib.makeScope newScope packages))).extend (selfNC: superNC: {})
+# Create exentsible scope and merge additional thirdparty apps
+in lib.recursiveUpdate ((lib.makeExtensible (_: (lib.makeScope newScope packages))).extend (selfNC: superNC: {})) (callPackage ./thirdparty.nix {})
```

Comment:
Can you please split this across multiple lines? This isn't very readable.

Also, the `.extend` doesn't make much sense here: especially since the `callPackage` is merged _after_ it.
Also, why recursiveUpdate and not `extend`?

Reply:
could you please help me with that? i copied this from the `default.nix` which is used by the wordpress plugins package set

Reply:
How about making `apps` itself extensible?

Then you could do `(makeExtensible (_: makeScope newScope packages)).extend (import ./thirdparty.nix)` which would be something along the lines of

```nix
_: { apps }: {
  apps = apps.extend (self: super: {
    foo = self.callPackage ./foo.nix {};
  });
}
```

Reply:
Thanks this was helpful 👍